next.config.js

```diff
@@ -1,3 +1,70 @@
-module.exports = {
-    useFileSystemPublicRoutes: false,
-}
\ No newline at end of file
+const path = require('path')
+
+const isProd = process.env.NODE_ENV === 'production'
+
+// paths
+const root = process.cwd()
+const src = path.resolve(root, './src')
+
+module.exports = {
+  useFileSystemPublicRoutes: false,
+  webpack(config, { isServer }) {
+    // Let webpack know about the theme alias to be able to use it in SCSS
+    // TS aliases should be set in the tsconfig
+    config.resolve.alias['theme'] = path.resolve(src, './theme')
+
+    // Add svgr support
+    config.module.rules.push({
+      // treat svgs in src/ as both react components and media files:
+      // https://create-react-app.dev/docs/adding-images-fonts-and-files/#adding-svgs
+      test: /\.svg$/,
+      exclude: /node_modules/,
+      use: [
+        {
+          loader: '@svgr/webpack',
+          options: {
+            memo: true,
+          },
+        },
+        {
+          loader: 'file-loader',
+          options: {
+            name: '[name].[hash:8].[ext]',
+            // next puts all their files in the _next folder, so follow the same standard
+            publicPath: `${
+              config.assetPrefix || config.basePath || ''
+            }/_next/static/media/`,
+            // On the server we the path is directly /static instead of /_next/static so account for that
+            outputPath: `${isServer ? '../' : ''}static/media/`,
+            // no need to copy the file if we're running in node --we just need the URI,
+            emitFile: !isServer,
+            esModule: config.esModule,
+          },
+        },
+      ],
+    })
+
+    // Add resjson support
+    config.module.rules.push({
+      // copy all resjsons to /media, preserving folder structure relative to src/
+      include: [/\.resjson$/],
+      loader: 'file-loader',
+      options: {
+        name: `${config.assetPrefix || config.basePath || ''}${
+          isProd ? '[path][name].[hash:8].[ext]' : '[path][name].[ext]'
+        }`,
+        // next puts all their files in the _next folder, so follow the same standard
+        publicPath: `${
+          config.assetPrefix || config.basePath || ''
+        }/_next/static/media/`,
+        // On the server we the path is directly /static instead of /_next/static so account for that
+        outputPath: `${isServer ? '../' : ''}static/media/`,
+        context: src,
+        // no need to copy the file if we're running in node --we just need the URI
+        emitFile: !isServer,
+      },
```

Comment:
We also need a catch-all file-loader for the other static assets we have ([equivalent of this](https://dev.azure.com/msazure/One/_git/azure-iots-saas?path=%2Fux-renderer%2Fwebpack.config.js&version=GBdevelop&line=385&lineEnd=386&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents))